### ISSUE TITLE
"{" in base contract constructor parameter string confuses instumentation

### ISSUE THREAD

**Issue by KaiRo-at** [sc-forks/solidity-coverage#478]
I have a contract using my patches in https://github.com/OpenZeppelin/openzeppelin-contracts/pull/2029 to add a URL to an ERC-1155 token, with code like this:

```
import ".../ERC1155MetadataURICatchAll.sol";
contract MYToken is ERC1155MetadataURICatchAll("https://example.com/meta/{id}"), {
    // Some contract code...
}
```

This compiles and works fine in general, but when running coverage, the instrumentation gets confused by the curly braces in the string and I get an error like this:
```
.../MYToken.sol:15:51: ParserError: Expected string end-quote.
 ... oken is ERC1155MetadataURICatchAll("https://example.com/meta/{event __CoverageMYToken(string fileName, uint256 lineNumber);
                                        ^--------------------------------------------------------------------------------------^

Compilation failed. See above.
Truffle v5.1.12 (core: 5.1.12)
Node v10.18.1
Cleaning up...
Event trace could not be read.
Error: ENOENT: no such file or directory, open './allFiredEvents'
Exiting without generating coverage...
```

When I remove the curly braces from the string before running coverage, it works correctly.

**Comment by KaiRo-at** [comment]
FWIW, if I move to a pattern where the parameter is noted with the constructor, e.g.
```
import ".../ERC1155MetadataURICatchAll.sol";
contract MYToken is ERC1155MetadataURICatchAll {
    constructor()
    ERC1155MetadataURICatchAll("https://example.com/CS2PS/meta/{id}")
    public
    {
    }
}
```
I get the same kind of error (message cropping not entirely correct as I had to edit out our actual URL given the project will not be public for some time):
```
.../MYToken.sol:56:32: ParserError: Expected string end-quote.
    ERC1155MetadataURICatchAll("https://example.com/meta ... yproject/contracts/MYToken.sol',1);
                               ^-------------------------------------------------------------------------^
```

**Comment by cgewecke** [comment]
@KaiRo-at Thanks for reporting! Nice catch. 

This should be fixed by #479. We just published yesterday - it could be a week (+) before another patch is released.

I think Zeppelin is migrating to solidity-coverage 0.7.x in the near term but at the moment they are using an older fork which is set up differently. 

There might be a bit of delay getting these changes to percolate through to your PR...

**Comment by KaiRo-at** [comment]
Awesome!
Just out of interest, does this cover the similar case in #issuecomment-586309919 as well?
(And FWIW, I'm using solidity-coverage 0.6.7 directly in our projects, I just in addition also contributed that ERC-1155 URL support in OpenZeppelin.)

**Comment by cgewecke** [comment]
@KaiRo-at Ah! Thanks for double-checking that...I didn't look at the second example closely enough. That will probably need it's own fix...

**Comment by cgewecke** [comment]
@KaiRo-at The second one is fixed in #480...

**Comment by cgewecke** [comment]
Published with `0.7.3`
